### DISCUSSION TITLE
`chdir` doesn't respect `.tflint.hcl` from the called directory

### DISCUSSION THREAD

**Discussion by amaxine** [terraform-linters/tflint#1797] [Q&A] [answered]
### Summary

Prior to the deprecation of accepting path arguments, we relied on keeping a single `.tflint.hcl` in the root of our Terraform configuration monorepo that got enforced across all configurations. With `chdir`, this is no longer the case. This is a problem for us, as the default rule set warns on certain behaviour we don't want - for example we enforce provider versions in modules only.

We would like to continue to have a singular configuration for a monorepo, which was possible prior to this change.

### Command

tflint --chdir=some/dir

### Terraform Configuration

```terraform
terraform {
  required_providers {
    datadog = {
      source = "DataDog/datadog"
    }
  }
}
```

### TFLint Configuration

```HCL
plugin "aws" {
  enabled = true
  version = "0.23.1"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
}

rule "terraform_unused_required_providers" {
  enabled = true
}

rule "terraform_naming_convention" {
  enabled = true
}

rule "terraform_required_providers" {
  enabled = false
}

rule "terraform_required_version" {
  enabled = false
}
```

### Output

```Shell
N/A
```

### TFLint Version

0.47.0

### Terraform Version

1.5.2

### Operating System

- [X] Linux
- [ ] macOS
- [ ] Windows

**Comment by amaxine** [comment]
I'm not sure why this was converted to a discussion, I'd consider this a regression compared to pre 0.44 releases of tflint. `chdir` as it stands isn't a 1 to 1 replacement of previous functionality.

**Comment by bendrucker** [comment, marked as answer]
See https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/working-directory.md, this is still possible. 

We made the decision that `chdir` should happen before anything meaningful, including loading configuration, and should be functionally equivalent to `cd $dir && tflint` in a shell. Terraform does the same. It is not a regression/bug.

**Reply by amaxine** [reply]
I think release notes should have clarified that. The recommendation to migrate to using `chdir` does not make it clear that the behaviour is different. Anyway, thanks for the pointer.

**Reply by bendrucker** [reply]
You also can use `git rev-parse --show-toplevel` from a Git repo depending on the desired behavior. You can always easily construct an absolute path to your config that TFLint will not attempt to resolve. 

If configuration paths were resolved relative to the original working directory, it would be impossible to implement the reverse and read the configuration from each recursed directory. It may not be the most convenient for you personally, but that's what makes it the clear best choice.

**Reply by bendrucker** [reply]
> In most cases the directory argument can be replaced with `--chdir`

Obviously this could be clearer about exactly which cases it's not 1:1, but there's a dedicated documentation page and this is a volunteer project. Sometimes you have to read the docs, that doesn't seem unreasonable here.
